tests/jlm/mlir/frontend/TestMlirToRvsdg.cpp

```diff
@@ -0,0 +1,127 @@
+    arguments.push_back(builder->getType<LoopStateEdgeType>());
+    ::llvm::ArrayRef argumentsArray(arguments);
+
+    // Handle function results
+    ::llvm::SmallVector<mlir::Type> results;
+    results.push_back(builder->getIntegerType(32));
+    results.push_back(builder->getType<IOStateEdgeType>());
+    results.push_back(builder->getType<MemStateEdgeType>());
+    results.push_back(builder->getType<LoopStateEdgeType>());
+    ::llvm::ArrayRef resultsArray(results);
+
+    // LambdaNodes return a LambdaRefType
+    ::llvm::SmallVector<mlir::Type> lambdaRef;
+    auto refType = builder->getType<LambdaRefType>(argumentsArray, resultsArray);
+    lambdaRef.push_back(refType);
+
+    // Add function attributes
+    ::llvm::SmallVector<mlir::NamedAttribute> attributes;
+    auto attributeName = builder->getStringAttr("sym_name");
+    auto attributeValue = builder->getStringAttr("test");
+    auto symbolName = builder->getNamedAttr(attributeName, attributeValue);
+    attributes.push_back(symbolName);
+    ::llvm::ArrayRef<::mlir::NamedAttribute> attributesRef(attributes);
+
+    // Add inputs to the function
+    ::llvm::SmallVector<mlir::Value> inputs;
+
+    // Create the lambda node and add it to the region/block it resides in
+    auto lambda =
+        builder->create<LambdaNode>(builder->getUnknownLoc(), lambdaRef, inputs, attributesRef);
+    omegaBlock->push_back(lambda);
+    auto & lambdaRegion = lambda.getRegion();
+    auto * lambdaBlock = new mlir::Block;
+    lambdaRegion.push_back(lambdaBlock);
+
+    // Add arguments to the region
+    lambdaBlock->addArgument(builder->getType<IOStateEdgeType>(), builder->getUnknownLoc());
+    lambdaBlock->addArgument(builder->getType<MemStateEdgeType>(), builder->getUnknownLoc());
+    lambdaBlock->addArgument(builder->getType<LoopStateEdgeType>(), builder->getUnknownLoc());
+
+    auto constOp = builder->create<mlir::arith::ConstantIntOp>(builder->getUnknownLoc(), 1, 32);
+    lambdaBlock->push_back(constOp);
+
+    ::llvm::SmallVector<mlir::Value> regionResults;
+    regionResults.push_back(constOp);
+    regionResults.push_back(lambdaBlock->getArgument(0));
+    regionResults.push_back(lambdaBlock->getArgument(1));
+    regionResults.push_back(lambdaBlock->getArgument(2));
+
+    // Handle the result of the lambda
+    auto lambdaResult = builder->create<LambdaResult>(builder->getUnknownLoc(), regionResults);
+    lambdaBlock->push_back(lambdaResult);
+
+    // Handle the result of the omega
+    ::llvm::SmallVector<mlir::Value> omegaRegionResults;
+    omegaRegionResults.push_back(lambda);
+    auto omegaResult = builder->create<OmegaResult>(builder->getUnknownLoc(), omegaRegionResults);
+    omegaBlock->push_back(omegaResult);
+
+    std::unique_ptr<mlir::Block> rootBlock = std::make_unique<mlir::Block>();
+    rootBlock->push_back(omega);
+
+    // Convert the MLIR to RVSDG
+    jlm::mlirrvsdg::MlirToRvsdg rvsdggen;
+    auto rvsdgModule = rvsdggen.convertMlir(rootBlock);
```

Comment:
Maybe you can add a static `CreateAndConvert()` method for convenience to the MlirToRvsdgConverter such that you do not always need to explicitly do these two lines.

Reply:
Added method